pulsar-client-cpp/tests/BasicEndToEndTest.cc

```diff
@@ -1323,7 +1324,14 @@ TEST(BasicEndToEndTest, testRSAEncryption) {
     std::string subName = "my-sub-name";
     Producer producer;
 
-    std::shared_ptr<EncKeyReader> keyReader = std::make_shared<EncKeyReader>();
+    std::string PUBLIC_CERT_FILE_PATH =
```

Comment:
Should `EncKeyReader` in `testEncryptionFailure` also be replaced with the default crypto key reader? Then the implementation of `EncKeyReader` could be removed.

Reply:
There is another test method `testEncryptionFailure` that needs to use this mock class, testEncryptionFailure mainly tests for some failure cases, it may be more appropriate to use this mock class

Reply:
But the implementation difference between `EncKeyReader` and `DefaultCryptoKeyReader` is just that `EncKeyReader` uses hard coding path while `DefaultCryptoKeyReader` does not, right?  The tests cases of  `testEncryptionFailure` are mocked by `prodConf.addEncryptionKey(path)`  but not the `EncKeyReader` itself.

Reply:
Thanks, I have removed this class `EncKeyReader`, PTAL @BewareMyPower